src/components/mainpage/sponsors/sponsor-section.js

```diff
@@ -0,0 +1,56 @@
+import React from "react"
+import SingleSponsor from "./single-sponsor"
+import sponsorsStyles from "../../../styles/sponsors.module.css"
+import Carousel from "../../utils/carousel"
+import { useSponsors } from "../../hooks/sponsors-query"
+
+export const getSponsors = (data, type) => {
+  return data.allMarkdownRemark.edges
+    .filter(({ node }) => node.frontmatter.type === type)
+    .map(({ node }) => (
+      <SingleSponsor
+        key={node.frontmatter.name}
+        sponsor={node.frontmatter}
+        type={type}
+      />
+    ))
+}
+
+const SponsorSection = ({ type }) => {
+  const data = useSponsors()
+
+  let wrapperType, sponsorText, sponsorType, numDesktop
+  if (type === "gold") {
+    wrapperType = sponsorsStyles.goldWrapper
+    sponsorText = sponsorsStyles.goldSponsor
+    sponsorType = "Gold"
+    numDesktop = 4
```

Comment:
You should assign this variable with the number of sponsors that come with the data query. This way you'll only have one assignment and if we change the number of sponsors in a category, which might happen because of the main sponsor situation, you won't have to go back to this code to change this number.